controllers/controllers.go

```diff
@@ -198,6 +198,15 @@ func isValidWindowsNode(o client.Object, byoh bool) bool {
 	return true
 }
 
+// getWindowsNodeVersions returns a list of all the WMCO versions that have last configured any nodes
+func getWindowsNodeVersions(nodes []core.Node) []string {
```

Comment:
I don't think this is necesary (the loop). However, I can see the benefit of the logic encapsulation for extracting the WMCO version from a node. 

In addition, `getWindowsNodeVersions` may read as "returns the Windows Server versions for all nodes".

Consider: 
```suggestion
func extractWMCOVersion(node []core.Node) string {
```

Reply:
I can get rid of this function/loop wholly by moving the logic into `isTiedToRelevantVersion()` as you suggested. Extracting the WMCO version from a node is a simple map access `node.Annotations[metadata.VersionAnnotation]` so no need to encapsulate